src/modules/lua_mtev.c

```diff
@@ -3252,13 +3252,12 @@ nl_hmac_sha1_encode(lua_State *L) {
   char encoded[29];
 
   if(lua_gettop(L) != 2) luaL_error(L, "bad call to mtev.hmac_sha1_encode");
-  encoded_len = 28; /* the length of the base64 encoded HMAC-SHA1 result will always be 28 */
 
   message = (const unsigned char *)lua_tolstring(L, 1, &messagelen);
   key = (const unsigned char *)lua_tolstring(L, 2, &keylen);
 
   HMAC(EVP_sha1(), key, keylen, message, messagelen, result, &md_len);
-  encoded_len = mtev_b64_encode(result, md_len, encoded, encoded_len);
+  encoded_len = mtev_b64_encode(result, md_len, encoded, mtev_b64_encode_len(md_len));
```

Comment:
why wouldn't this be encoded_len?  It's the result of ``mtev_b64_encode_len``

Reply:
`encoded_len` is uninitialized here. It used to be hard coded to 28, but I removed that in favor of calling the function.